Fixes bug 101220: supports pasting text into multiple selections.
This has also been accepted into Scintilla:
bug https://sourceforge.net/p/scintilla/bugs/1541/ , to appear
in the next version after 3.3.6
Index: cocoa/ScintillaCocoa.mm
===================================================================
--- cocoa/ScintillaCocoa.mm	(revision 82776)
+++ cocoa/ScintillaCocoa.mm	(working copy)
@@ -938,29 +938,29 @@
   if (forceRectangular)
     selectedText.rectangular = forceRectangular;
   
   if (!ok || selectedText.Empty())
     // No data or no flavor we support.
     return;
   
   pdoc->BeginUndoAction();
   ClearSelection(false);
   int length = selectedText.Length();
+  SelectionPosition selStart = sel.RangeMain().Start();
   if (selectedText.rectangular)
   {
-    SelectionPosition selStart = sel.RangeMain().Start();
     PasteRectangular(selStart, selectedText.Data(), length);
   }
-  else 
-    if (pdoc->InsertString(sel.RangeMain().caret.Position(), selectedText.Data(), length))
-      SetEmptySelection(sel.RangeMain().caret.Position() + length);
-  
+  else
+  {
+    InsertPaste(selStart, selectedText.Data(), length);
+  }
   pdoc->EndUndoAction();
   
   Redraw();
   EnsureCaretVisible();
 }
 
 //--------------------------------------------------------------------------------------------------
 
 void ScintillaCocoa::CTPaint(void* gc, NSRect rc) {
 #pragma unused(rc)
